File: flutter_app/generate_icon.py
"""
Generates the SalamaRecover launcher icon (1024x1024 PNG).
Run once: python generate_icon.py
Requires: pip install pillow
"""
from PIL import Image, ImageDraw
import math, os

SIZE = 1024
img = Image.new('RGBA', (SIZE, SIZE), (0, 0, 0, 0))
draw = ImageDraw.Draw(img)

# Blue gradient circle background (simulate with solid + lighter overlay)
draw.ellipse([0, 0, SIZE, SIZE], fill='#0077B6')

# Slightly lighter arc at top-left for depth
for r in range(60):
    alpha = int(60 * (1 - r / 60))
    draw.ellipse([r, r, SIZE // 2 + 80 - r, SIZE // 2 + 80 - r],
                 outline=(255, 255, 255, alpha), width=1)

# White heart outline
cx, cy = SIZE // 2, SIZE // 2 + 30
heart_scale = 3.2
pts = []
for angle in range(0, 360):
    t = math.radians(angle)
    x = heart_scale * 16 * (math.sin(t) ** 3)
    y = -heart_scale * (13 * math.cos(t) - 5 * math.cos(2*t)
                        - 2 * math.cos(3*t) - math.cos(4*t))
    pts.append((cx + x, cy + y))

draw.polygon(pts, fill='white')

# Blue medical cross over the heart
cross_color = '#0077B6'
bar_w = 80
bar_h = 260
draw.rectangle([cx - bar_w//2, cy - bar_h//2, cx + bar_w//2, cy + bar_h//2],
               fill=cross_color)
draw.rectangle([cx - bar_h//2, cy - bar_w//2, cx + bar_h//2, cy + bar_w//2],
               fill=cross_color)

# Green accent dot bottom-right
draw.ellipse([SIZE - 220, SIZE - 220, SIZE - 80, SIZE - 80], fill='#00B37E')

out = os.path.join('assets', 'icons', 'launcher_icon.png')
img.save(out)
print(f'Icon saved to {out}')

File: flutter_app/generate_icon.dart
/// Generates assets/icons/launcher_icon.png
/// Run from flutter_app/: dart run generate_icon.dart
/// Then: dart run flutter_launcher_icons
///
/// Design: blue (#0077B6) background, white medical cross (rounded ends),
///         green (#00B37E) pulse dot — SalamaRecover brand colours.
import 'dart:io';
import 'package:image/image.dart' as img;

void main() {
  const size = 1024;
  const cx = size ~/ 2;
  const cy = size ~/ 2;

  // Brand colours
  final blue  = img.ColorRgba8(0, 119, 182, 255);   // #0077B6
  final white = img.ColorRgba8(255, 255, 255, 255);
  final green = img.ColorRgba8(0, 179, 126, 255);    // #00B37E

  final image = img.Image(width: size, height: size);

  // ── 1. Solid blue background ────────────────────────────────
  img.fill(image, color: blue);

  // ── 2. White medical cross (rounded-end bars) ───────────────
  //  Bar dimensions: arm width = 13 %, arm length = 46 % of icon
  final arm  = (size * 0.46).round(); // half-length of each arm
  final half = (size * 0.065).round(); // half-width of bar

  // Vertical bar
  img.fillRect(image,
      x1: cx - half, y1: cy - arm, x2: cx + half, y2: cy + arm,
      color: white);
  // Round the top and bottom ends
  img.fillCircle(image, x: cx, y: cy - arm, radius: half, color: white);
  img.fillCircle(image, x: cx, y: cy + arm, radius: half, color: white);

  // Horizontal bar
  img.fillRect(image,
      x1: cx - arm, y1: cy - half, x2: cx + arm, y2: cy + half,
      color: white);
  // Round the left and right ends
  img.fillCircle(image, x: cx - arm, y: cy, radius: half, color: white);
  img.fillCircle(image, x: cx + arm, y: cy, radius: half, color: white);

  // ── 3. Green pulse dot (top-right of centre cross) ──────────
  //  Placed where the right arm meets the top arm — like a heartbeat indicator
  final dotR   = (size * 0.075).round();
  final dotOff = (size * 0.30).round();
  img.fillCircle(image,
      x: cx + dotOff, y: cy - dotOff, radius: dotR, color: green);

  // ── 4. Save ─────────────────────────────────────────────────
  final outFile = File('assets/icons/launcher_icon.png');
  outFile.writeAsBytesSync(img.encodePng(image));

  print('✓  Icon saved → ${outFile.path}  ($size × $size px)');
  print('   Now run:  dart run flutter_launcher_icons');
}
